pkg/workflows/utils.go

```diff
@@ -0,0 +1,21 @@
+package workflows
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+)
+
+func GenerateExecutionID(workflowID, eventID string) (string, error) {
```

Comment:
nit/ `Generate` feels like a strong word for hash + hex encode. What about something like `Encode`?
```suggestion
func EncodeExecutionID(workflowID, eventID string) (string, error) {
```
Also, you can probably eliminate the error from the signature if you wanted to, since the hash impl used does not return errors and is just satisfying the interface.

Reply:
For comparison:
```suggestion
func NewExecutionID(workflowID, eventID string) string {
```
```suggestion
func EncodeExecutionID(workflowID, eventID string) string {
```

Reply:
```suggestion
func DeriveExecutionID(workflowID, eventID string) string {
```